webservice/model/assignment.go

```diff
@@ -138,22 +139,32 @@ func (repo *AssignmentRepository) GetAllToUserSorted(UserID int) ([]Assignment,
 // Insert a new assignment to the database
 func (repo *AssignmentRepository) Insert(assignment Assignment) (int, error) {
 
+	tx, err := db.GetDB().Begin() //start transaction
+	if err != nil {
+		log.Println(err.Error())
+		return 0, err
+	}
+
 	// Get current Norwegian time in string format TODO time-norwegian
 	date := util.ConvertTimeStampToString(util.GetTimeInCorrectTimeZone())
 
 	// Create query string
 	query := "INSERT INTO assignments (name, description, publish, deadline, course_id) VALUES (?, ?, ?, ?, ?);"
 	// Prepare and execute query
-	ex, err := db.GetDB().Exec(query, assignment.Name, assignment.Description, assignment.Publish, assignment.Deadline, assignment.CourseID)
+	ex, err := tx.Exec(query, assignment.Name, assignment.Description, assignment.Publish, assignment.Deadline, assignment.CourseID)
 	// Check for error
 	if err != nil {
+		log.Fatal(err.Error())
```

Comment:
No need for logging inside here, cuase error is a return value.
And don't use log.Fatal() cauase it will stop the whole application.